### PR TITLE
Feature: Allow vote to be recasted

### PR DESCRIPTION
**Pull request by RakeshUP**
Task: [APP-1443](https://aragonassociation.atlassian.net/browse/APP-1443)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

[APP-1443]: https://aragonassociation.atlassian.net/browse/APP-1443?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Fabricevladimir — APPROVED** [review]
Looking good to me

**Review by sepehr2github — APPROVED** [review]
Thanks, LGTM 👍

File: packages/web-app/src/pages/proposal.tsx [hunk 1]
```diff
@@ -484,6 +486,7 @@ const Proposal: React.FC = () => {
     t,
     voted,
     i18n.language,
+    canVote,
```
Comment (WissenIstNacht) [inline packages/web-app/src/pages/proposal.tsx:489]: `canVote` is no longer used outside of this `useMemo`. You could technically move it into this `useMemo`.

File: packages/web-app/src/pages/proposal.tsx [hunk 2]
```diff
@@ -435,7 +435,9 @@ const Proposal: React.FC = () => {
       return [voteStatus, voteButtonLabel];
 
     voteButtonLabel = voted
-      ? t('votingTerminal.status.voteSubmitted')
+      ? canVote
+        ? t('votingTerminal.status.revote')
+        : t('votingTerminal.status.voteSubmitted')
       : t('votingTerminal.voteOver');
```
Comment (WissenIstNacht) [inline packages/web-app/src/pages/proposal.tsx:441]: If `voted` is false, then we display that the vote is over. But when I look at how `voted` is computed it doesn't seem to have any information about the vote duration.

**Review by WissenIstNacht — COMMENTED** [review]
Oh, I just realized Sepehr's review came in in the meantime* :(

Still leaving them here, feel free to implement changes if you agree or merge if you don't.

Edit: And that you actually merged the PR half an hour ago lol.

**Pull request merged**